src/widget.cpp

```diff
@@ -2564,24 +2564,28 @@ NWidgetLeaf::NWidgetLeaf(WidgetType tp, Colours colour, WidgetID index, uint32_t
 			this->SetFill(0, 0);
 			this->SetMinimalSize(WidgetDimensions::WD_STICKYBOX_WIDTH, WidgetDimensions::WD_CAPTION_HEIGHT);
 			this->SetDataTip(STR_NULL, STR_TOOLTIP_STICKY);
+			this->SetAspect((float)this->min_x / (float)this->min_y);
```

Comment:
Shouldn't these cast use `static_cast<float>` to prevent future issues?

Reply:
Added an additional SetAspect() method to handle this without adding casts in lots of places :)